WordPress/gutenberg
# Add Featured Image, Title and Excerpt to top of the settings sidebar

## What?
- Moves the Featured Image picker to the top of the sidebar.
    - Uses a new dropdown-based flow for selecting a featured image.
    - Adds the ability to upload a featured image using the native file picker.
- Adds new title and excerpt fields to the top of the sidebar.
- Removes the Featured Image and Excerpt panels.

## Why?
Part of https://github.com/WordPress/gutenberg/issues/39082.

## How?
- The Featured Image changes are made in-place using the existing `PostFeaturedImage` component in `@wordpress/editor`.
    - This component is somewhat extensible using a filter that third parties can hook into. I took care to not remove any props that third parties may be using, kept all CSS class names the same, and checked that the new styles do not affect elements that third parties may be adding here.
    - I added the _Upload_ button using `FormFileUpload`.
    - This component was getting very indented so I refactored it into some smaller components. 
- The title and excerpt fields are a new component called `PostSummary` in `@wordpress/edit-post`.
    - It doesn't make sense I think to put these in `@wordpress/editor` as the design we're using is pretty tailored to the post editor sidebar. 
    - Both fields use `EditableText` as rich text formatting is not supported here.
- I removed the Featured Image and Excerpt components from `@wordpress/edit-post` but not `@wordpress/editor` as third parties may be using them.

## Testing Instructions
1. Create or edit a new post.
2. Add a featured image, title, excerpt, whatever!
3. Run the sample code in https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-featured-image/README.md to check that extending the Featured Image still works. 
4. Log in as a contributor to check that these fields don't appear.

## Screenshots or screencast 
https://user-images.githubusercontent.com/612155/178655097-8f8c2123-3e95-4423-b5a6-45b8760a5670.mp4
